v1/related.js

```diff
@@ -11,59 +11,89 @@ class Related {
         this._options = options;
     }
 
+    _relatedPageRedirect(hyper, domain, title) {
+        const params = `action=query&format=json&titles=${title}&redirects=1&converttitles=1`;
+        const apiURL = `https://${domain}/w/api.php?${params}`;
+        return hyper.get({
+            uri: apiURL,
+            headers: {
+                'content-type': 'application/json'
+            }
+        }).then((res) => {
+            return res.body;
+        }).catch((e) => {
+            hyper.logger.log('error/related', {
+                message: 'Failed to fetch related pages',
+                error: e
+            });
+            throw e;
+        });
+    }
+
     getPages(hyper, req) {
         const rp = req.params;
-        const rh = Object.assign({}, req.headers);
-        // we don't store /page/related no need for cache-control
-        if (rh['cache-control']) {
-            delete rh['cache-control'];
-        }
 
-        return hyper.post({
-            uri: new URI([rp.domain, 'sys', 'action', 'query']),
-            body: {
-                format: 'json',
-                generator: 'search',
-                gsrsearch: `morelike:${rp.title}`,
-                gsrnamespace: 0,
-                gsrwhat: 'text',
-                gsrinfo: '',
-                gsrprop: 'redirecttitle',
-                gsrlimit: 20
+        return this._relatedPageRedirect(hyper, rp.domain, rp.title).then((res) => {
+
+            // Return the right title to redirect if it has another language option
+            if (res.query.converted) {
```

Comment:
This will cause an error if `res` is empty. I'd suggest additional checks here and below to see if `res` and `res.query` exist. Similar to line 90 below

Reply:
Done.